tutorials/using-edge/production-deploy-to-ase-gpu.ipynb

```diff
@@ -0,0 +1,458 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.core.workspace import Workspace\n",
+    "\n",
+    "ws = Workspace.from_config()\n",
+    "print(ws.name, ws.resource_group, ws.location, ws.subscription_id, sep=\"\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Download the model\n",
+    "\n",
+    "Prior to registering the model, you should have a TensorFlow [Saved Model](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/saved_model/README.md) in the `resnet50` directory. This cell will download a [pretrained resnet50](http://download.tensorflow.org/models/official/20181001_resnet/savedmodels/resnet_v1_fp32_savedmodel_NCHW_jpg.tar.gz) and unpack it to that directory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import requests\n",
+    "import shutil\n",
+    "import tarfile\n",
+    "import tempfile\n",
+    "\n",
+    "from io import BytesIO\n",
+    "\n",
+    "model_url = \"http://download.tensorflow.org/models/official/20181001_resnet/savedmodels/resnet_v1_fp32_savedmodel_NCHW_jpg.tar.gz\"\n",
+    "\n",
+    "archive_prefix = \"./resnet_v1_fp32_savedmodel_NCHW_jpg/1538686758/\"\n",
+    "target_folder = \"resnet50\"\n",
+    "\n",
+    "if not os.path.exists(target_folder):\n",
+    "    response = requests.get(model_url)\n",
+    "    archive = tarfile.open(fileobj=BytesIO(response.content))\n",
+    "    with tempfile.TemporaryDirectory() as temp_folder:\n",
+    "        archive.extractall(temp_folder)\n",
+    "        shutil.copytree(\n",
+    "            os.path.join(temp_folder, archive_prefix), target_folder\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Register the model\n",
+    "Register an existing trained model, add description and tags."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.core.model import Model\n",
+    "\n",
+    "model = Model.register(\n",
+    "    model_path=\"resnet50\",  # This points to the local directory to upload.\n",
+    "    model_name=\"resnet50\",  # This is the name the model is registered as.\n",
+    "    tags={\"area\": \"Image classification\", \"type\": \"classification\"},\n",
+    "    description=\"Image classification trained on Imagenet Dataset\",\n",
+    "    workspace=ws,\n",
+    ")\n",
+    "\n",
+    "print(model.name, model.description, model.version)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Deploy the model as a web service to Edge\n",
```

Comment:
this code should live in `code/deploy`, not be generated via notebook writefile cell